src/components/App.jsx

```diff
@@ -1,9 +1,12 @@
 import React from "react";
 import ContactsApp from "./ContactsApp";
+import { withContacts } from "./ContactsList";
+
+const ConnectedContactsApp = withContacts(ContactsApp);
```

Comment:
I move up the data connection to make data shared with every part of the app without a central global state (like in redux architecture).

Reply:
Il y a un inconvénient toutefois : regrouper tout ce qui est gestion de données (y compris updates) dans le composant "racine" va provoquer des rerenders sur l'ensemble de l'app. Tu as fait ça en attendant cozy-client (qui a une approche plus "on connecte le composant qui va bien où qu'il soit dans l'arbre) ou tu voulais essayer cette façon de faire et voir comment ça évolue ?

Reply:
J'ai fait ça car on n'avait pas besoin de redux. C'est effectivement le souci quand on prend une architecture à base de composant App : c'est l'App qui gère toutes les données et qui fait ruisseler les données vers les composants enfants. C'est dans ce cadre que l'utilisation de PureComponent et shouldComponentUpdate est utile pour soulager les rerendering.

Cela étant dit, avec `cozy/cozy-client` (ou toute autre lib de gestion d'état), nous pourrons bien évidemment brancher les composants pour que chacun s'inscrive à la partie des données qui l'intéresse et ne se rerender pas pour rien.

Rassembler tout au même endroit sans archi avec gestion d'état permet de faire ce que présente Abramov sur egghead avec son idiomatic redux : un refactoring simple et direct.